packages/ui-extensions/src/types/Stack.ts

```diff
@@ -0,0 +1,14 @@
+type Alignment = 'leading' | 'center' | 'trailing' | 'baseline';
+
+type Spacing =
+  | 'none'
+  | 'extraTight'
+  | 'tight'
+  | 'base'
+  | 'loose'
+  | 'extraLoose';
+
+export interface BaseStackProps {
+  alignment?: Alignment;
```

Comment:
> It's not clear to me what axis alignment is relative to

Totally agree. I think it made sense when these components only aligned their children on the main axis. Now that we introduced cross axis alignment on InlineStack (`blockAlignment` prop), having `alignment` doesn't make sense anymore.

> you didn't like that a stack had props to control alignment on both axis

Yeah, I wasn't convinced at first. I thought it was removing some of the simplicity in our API, but after some though, you and Tim convinced me its a good addition and a needed one.

> but since we do not have more than one row for stacks

Thats the goal, but along the way we added `wrap` for InlineStack. Something we should not expose in Argo as it will be removed in favour of a better Tiles. Its currently only used for discount tags in Checkout app.

Reply:
```
interface BlockStack {
  blockAlignment?: BlockAlignment;
  inlineAlignment?: InlineAlignment;
  spacing?: Spacing;
}
```

We could also use

```
interface BlockStack {
  alignment?: Alignment;
  crossAlignment?: CrossAlignment;
  spacing?: Spacing;
}
```
or
```
interface BlockStack {
  mainAlignment?: MainAlignment;
  crossAlignment?: CrossAlignment;
  spacing?: Spacing;
}
```

Reply:
@jffortier I like the first suggestion. It seems the most clear in either block or inline context.